mirrord/layer/src/file/hooks.rs

```diff
@@ -148,21 +158,29 @@ pub(super) unsafe extern "C" fn open_nocancel_detour(
 #[cfg(not(all(target_os = "macos", target_arch = "aarch64")))]
 #[hook_guard_fn]
 pub(super) unsafe extern "C" fn opendir_detour(raw_filename: *const c_char) -> usize {
-    open_logic(raw_filename, O_RDONLY, O_DIRECTORY)
-        .and_then(|fd| match fdopendir(fd) {
-            Detour::Success(success) => Detour::Success(success),
-            Detour::Bypass(bypass) => Detour::Bypass(bypass),
-            Detour::Error(fail) => {
-                close_layer_fd(fd);
-                Detour::Error(fail)
+    get_runtime()
+        .block_on(async {
+            match open_logic(raw_filename, O_RDONLY, O_DIRECTORY).await {
```

Comment:
You can keep `and_then` here, drop the `await` and use the `futures::TryStreamExt` (or whatever it's called).